### PR TITLE
NVMf Tests

### PR DESCRIPTION
**Pull request by narasimhan-v**
The following NVMf tests are added:
* target config
* nvmf discover
* nvmf connect
* nvmf disconnect
* nvmf connect cfg
* nvmf disconnect cfg
* clear target config

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

File: io/nvmf/nvmftest.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    @staticmethod
+    def nvme_devs_count():
+        """
+        Returns count of nvme devices in the system
+        """
+        cmd = "nvme list"
+        output = process.system_output(cmd, shell=True, ignore_status=True)
+        count = len(output.splitlines()) - 2
+        print "Count: %d" % count
```
Comment (PraveenPenguin) [inline io/nvmf/nvmftest.py]: can this be in `self.log.info `

File: io/nvmf/nvmftest.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+"""
+NVM Express over Fabrics defines a common architecture that supports a
+range of storage networking fabrics for NVMe block storage protocol over
```
Comment (PraveenPenguin) [inline io/nvmf/nvmftest.py]: `storage network fabrics` look me more appropriate

File: io/nvmf/nvmftest.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+a storage networking fabric.
```
Comment (PraveenPenguin) [inline io/nvmf/nvmftest.py]: here as well

File: io/nvmf/nvmftest.py (hunk 2 above)
Comment (narasimhan-v) [reply io/nvmf/nvmftest.py]: @PraveenPenguin This is from: http://www.nvmexpress.org/wp-content/uploads/NVMe_Over_Fabrics.pdf

File: io/nvmf/nvmftest.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    def setUp(self):
+        """
+        Sets up NVMf configuration
+        """
+        self.nvme_ns = self.params.get('namespace', default='')
+        self.peer_ip = self.params.get('peer_ip', default='')
+        if not self.nvme_ns:
+            self.cancel("No namespace provided")
+        if not self.peer_ip:
+            self.cancel("No peer_ip provided")
```
[the remark this answers is not in the document]
Comment (narasimhan-v) [reply io/nvmf/nvmftest.py]: I was thinking of giving clear message to user on which parameter is not given. I will change it.

File: io/nvmf/nvmftest.py (hunk 1 above)
Comment (narasimhan-v) [reply io/nvmf/nvmftest.py]: @PraveenPenguin Thanks for catching this. I need to remove this.

File: io/nvmf/nvmftest.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        smm = SoftwareManager()
+        if not smm.check_installed("nvme-cli") and not \
+                smm.install("nvme-cli"):
+            self.cancel('nvme-cli is needed for the test to be run')
+        if not linux_modules.module_is_loaded("nvme-rdma"):
+            linux_modules.load_module("nvme-rdma")
```
[the remark this answers is not in the document]
Comment (narasimhan-v) [reply io/nvmf/nvmftest.py]: Sure, will do.

File: io/nvmf/nvmftest.py (hunk 2 above)
Comment (PraveenPenguin) [reply io/nvmf/nvmftest.py]: ah sure i just go with how we pronounce

Comment (narasimhan-v) [reply io/nvmf/nvmftest.py]: Agreed, but since this is how it is in the community, I will stick with this :-)

**Comment by narasimhan-v** [comment]
@PraveenPenguin comments taken care of. Please have a check now.

**Pull request merged**